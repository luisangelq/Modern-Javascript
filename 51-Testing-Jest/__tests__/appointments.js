import Citas from '../js/classes/Citas.js';

describe('test appointment Class', () => {
    const citas = new Citas();

    const id = Date.now();

    test('Add new appointment', () => {
        const citaObj = {
            id,
            mascota: "Hook",
            propietario: "Luis",
            telefono: "3326264356",
            fecha: "2018-01-01",
            hora: "10:00",
            sintomas: "No duerme",
        };

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();   

    })
    
    test('Update appointment', () => {
        const citaUpdated = {
            id,
            mascota: "New Pet",
            propietario: "Luis",
            telefono: "3326264356",
            fecha: "2018-01-01",
            hora: "10:00",
            sintomas: "No duerme",
        };
        citas.editarCita(citaUpdated);

        expect(citas).toMatchSnapshot();
    })

    test('Delete appointment', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
})