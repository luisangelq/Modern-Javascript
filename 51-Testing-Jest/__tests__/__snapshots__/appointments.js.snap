// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test appointment Class Add new appointment 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "2018-01-01",
      "hora": "10:00",
      "id": 1629766490402,
      "mascota": "Hook",
      "propietario": "Luis",
      "sintomas": "No duerme",
      "telefono": "3326264356",
    },
  ],
}
`;

exports[`test appointment Class Delete appointment 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`test appointment Class Update appointment 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "2018-01-01",
      "hora": "10:00",
      "id": 1629766490402,
      "mascota": "New Pet",
      "propietario": "Luis",
      "sintomas": "No duerme",
      "telefono": "3326264356",
    },
  ],
}
`;
